### PR TITLE
[MINOR][DOCS][CONNECT] Fix docs to run Spark Connect locally built

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR adds some more command examples to run Spark Connect that you built locally.

### Why are the changes needed?

To guide developers to run Spark Connect they built locally.

### Does this PR introduce _any_ user-facing change?

No, dev-only and doc-only.

### How was this patch tested?

The commands were tested manually in my local.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @zhengruifeng @amaliujia FYI

**Comment by zhengruifeng** [comment]
maybe we can also modify the last test cmd:

`./run-tests --testnames 'pyspark.sql.tests.test_connect_basic'`

to 

`./python/run-tests --testnames 'pyspark.sql.tests.test_connect_basic'`

File: python/pyspark/sql/connect/README.md [hunk 1]
```diff
@@ -9,22 +9,34 @@ of Spark. To enable it, you only need to activate the driver plugin for Spark Co
...
+./bin/pyspark \
+  --jars `ls connector/connect/target/**/spark-connect*SNAPSHOT.jar | paste -sd ',' -` \
+  --conf spark.plugins=org.apache.spark.sql.connect.SparkConnectPlugin
+```
+
+To use the release version of Spark Connect:
+
 ```bash
 ./bin/spark-shell \
   --packages org.apache.spark:spark-connect_2.12:3.4.0 \
```
Comment (grundprinzip) [inline python/pyspark/sql/connect/README.md:42]: Is this the preferred way of doing it?

**Review by grundprinzip — APPROVED** [review]

**Review by zhengruifeng — APPROVED** [review]

**Review by amaliujia — APPROVED** [review]

File: python/pyspark/sql/connect/README.md (hunk 1 above)
Comment (HyukjinKwon) [reply python/pyspark/sql/connect/README.md:42]: Yup, the user-facing documentation will only have this alone.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**